File: TestSortBySelection/Python/TestSortBySelection.py
#TestSortBySelection.py
#19/02/2021

# findMin(array)
# Description: Find the index for the array position with less value
def findMin(array):
	minor = array[0]
	minIndex = 0
	for i in range(1, len(array)):
		if array[i] < minor:
			minor = array[i]
			minIndex = i
	return minIndex

# sortBySelection(array)
# Description: Return the input array sorted from low to high values 	
def sortBySelection(array):
	sortedArray = []
	for i in range(len(array)):
		minIndex = findMin(array)
		sortedArray.append(array.pop(minIndex))
	return sortedArray

# TEST
sortedArray = sortBySelection([5, 3, 6, 2, 10])
print(sortedArray)		
	

File: TestSortBySelection/C/TestSortBySelection.c
//TestSortBySelection.c
//19/02/2021
#include <stdio.h>

void swap(double *xp, double *yp) 
{ 
    double temp = *xp; 
    *xp = *yp; 
    *yp = temp; 
}; 

//int sortBySelection(double array[], int sizeArray)
//Description: Return the input array sorted from low to high values 	
void sortBySelection(double array[], int sizeArray){
	int i, j, minIndex;
	
	for(i = 0; i < sizeArray - 1; i++){
		
		minIndex = i;
		for(j = i+1; j < sizeArray; j++){
			if (array[j] < array[minIndex]) 
				minIndex = j;
		}		
		swap(&array[minIndex], &array[i]);
	}	
};
	
// TEST
int main(){
	double array[5] = {5, 3, 6, 2, 10};
	int sizeArray = sizeof(array)/sizeof(double);
	
	/*Sort the array*/
	sortBySelection(array, sizeArray);
	
	/*Plot sorted array*/
	for(int i = 0; i < sizeArray; i++)
		printf("%f ", array[i]);
	
	return 0;
}	
